server/recurring/cluster.go

```diff
@@ -0,0 +1,305 @@
+package recurring
+
+import (
+	"math"
+	"sort"
+	"strings"
+)
+
+var (
+	specialWeights = map[string]float64{
+		"amazon": 10,
```

Comment:
Add a `pwp` multiplier for `0`. This will make it so the word is essentially useless.

Also add something where if a words value is 0 we maybe exclude it from the vector entirely?